experimental/GModule/Cohomology.jl

```diff
@@ -474,7 +474,7 @@ export action, cohomology_group, extension, pc_group_with_isomorphism
 export induce, is_consistent, istwo_cocycle, all_extensions
 export split_extension, extension_with_abelian_kernel
 
-Oscar.dim(C::GModule) = torsion_free_rank(C.M)
+Oscar.dim(C::GModule) = rank(C.M)
```

Comment:
There is no `rank(::FinGenAbGroup` anymore in conjunction with https://github.com/thofma/Hecke.jl/pull/1224.

Reply:
See the discussion above. `C.M` can also be a `FreeModule` (or something).

Reply:
I find this `dim` function rather problematic to start with, there is a reason we use `rank` for modules in general... but that's besides the point right now